### DISCUSSION TITLE
Revealjs: slide header with section and subsection

### DISCUSSION THREAD

**Discussion by fbob** [quarto-dev/quarto-cli#2273] [Q&A] [answered]
Hello,
I would like to reproduce a layout from latex beamer using quarto and revealjs which puts the current section and subsection of a presentation a the top of the slide, the slide title being just underneath. Before each new subsection a new slide is displayed showing the section and subsections titles.

See screenshots below:

![image](https://user-images.githubusercontent.com/8211124/188202443-f16aa311-7b8a-4f96-87a3-c1009a1ed62d.png)

![image](https://user-images.githubusercontent.com/8211124/188202381-8fb28439-4d42-49ab-941d-3d2113f9a524.png)

I'm very new to quarto and revealjs and need some help.
Any revaljs guru here that can help ?

Thanks !
Have a nice day.

**Comment by mcanouil** [comment]
This requires "some" CSS, possibly a modification of <https://github.com/andrie/reveal-auto-agenda>, but most likely this would be a format extension <https://quarto.org/docs/extensions/formats.html>.

**Reply by fbob** [reply]
Thank you very much @mcanouil for your kind and helpful answer !
The [reveal-auto-agenda](https://github.com/andrie/reveal-auto-agenda) quarto extension works very well, thanks.

I think the main difficulty for me is to find some .scss examples that can show me how to place the section and subsection on the upper left and right of the slides. I have no experience with css at all.

Anyway, many thanks for your kind help :+1:

**Reply by mcanouil** [reply]
Not perfect, but here is an example only with css:

```css
.reveal h2 {
  text-align: left;
  vertical-align: top;
}

.reveal h3 {
  text-align: right;
  position: inherit;
  top: 0;
  right: 0;
}
```

```
---
title: "Untitled"
format:
  revealjs:
    css: theme.css
---

## Section

### Sub section

content
```

<img width="1502" alt="image" src="https://user-images.githubusercontent.com/8896044/188318009-be066d36-0d60-4aac-b733-b65a54afe78f.png">

**Reply by mcanouil** [reply]
For more, checkout https://revealjs.com/.

**Reply by fbob** [reply]
Many thanks @mcanouil, it's very nice from you :+1: 
I've played with it a lite bit and  I came across a strange behavior. I think it's me who doesn't know how to use reveal well but it's also possible that it's just not feasible to do that.

I saw that you used levels h2 for the sections, h3 for the sub-sections and h4 for the slide's title.

I've changed the theme.css to try to have the sections and sub-sections in h1 and h2 but it doesn't work

Another difficulty and that I can't start a new slide for each level of header corresponding to the slides keeping the section and subsection titles at the top of the slide.

I've made some screenshots to try to better explain the issues/behaviors.

**Strang behavior with sections and subsections as h1 and h2 levels** 

```css
.reveal h1 {
  text-align: left;
  vertical-align: top;
}

.reveal h2 {
  text-align: right;
  position: inherit;
  top: 0;
  right: 0;
}
```
```md
---
title: "Playing with the layout"
format:
  revealjs:
    css: theme.css
---

# Section

## Subsection

### Slide title

content of the slide
```

![image](https://user-images.githubusercontent.com/8211124/188928589-89edae5a-6bfa-4f74-a17c-84ad49de1c8a.png)

**Strange behavior with  `slide-level: 3`**

```md
---
title: "Playing with the layout"
format:
  revealjs:
    css: theme.css
    slide-level: 3
---

# Section

## Subsection

### Slide title

Content of the first slide

### Another slide

Content of the second slide
```
![image](https://user-images.githubusercontent.com/8211124/188932391-b989c0e2-c42b-47f5-94e8-5b9a375cd7ef.png)

You can see each section and subsection starts a new slide and that the layout is lost i.e the section and subsection are not on the current slide.

I'm sorry if this seems trivial to you or if you find my questions stupid but I couldn't find explanations in the revealjs's doc, maybe I didn't look in the right place.

Have fun and good day to all.

**Reply by mcanouil** [reply]
To be clear about the vocabulary and how Reveal.js defines headers:

- Level 1 headers are more like "chapters".
- Level 2 headers are slide titles.
- Level 3 Headers are slides subtitles.

(I am AFK right now, so can't really do much more.)

**Reply by mcanouil** [reply]
Some update, the layout you try to achieve is not really trivial because Reveal.js is building most of it on it's own.
The best would be for you to play with CSS or even write your own modified Reveal.js Pandoc template (see <https://github.com/quarto-dev/quarto-cli/blob/main/src/resources/formats/revealjs/pandoc/template.revealjs>).

**Comment by jimjam-slam** [comment]
You could also write a Lua filter to add the section/subsection content as well! It'd just need to walk the sections, walk the slides inside adding classed spans, then you use some basic CSS to position the spans.

**Reply by mcanouil** [reply]
The Lua filter is probably the best option once you know and have defined the required CSS classes.
But, to get properly the two side by side properly, this involves also to tweak the content of the slide to add some margin at the top, to avoid content to overlap with the slide part/title/subtitle.

**Reply by fbob** [reply]
Thanks to both of you for your quick feedback.
I realize that it's not that trivial and that it requires delving into more advanced notions of Revealjs.

Thanks for your help and advice, it's really appreciated !

**Comment by fbob** [comment, marked as answer]
For those who may be interested, there is a new extension done by @shafayetShafee 
 who does this :
https://github.com/shafayetShafee/reveal-header
